From 17af3ebfa976f2da08b6f8c9ae12cc3fd2513026 Mon Sep 17 00:00:00 2001
From: Lu Yahan <yahan@iscas.ac.cn>
Date: Mon, 10 Jul 2023 11:11:06 +0800
Subject: [PATCH] [riscv][compiler] Generalize InstructionSelectorT for
 Turboshaft (part 3)

Bug: v8:12783

Change-Id: Iad894ecc8ee7fd071ba971f5caeec21320dcf9d9
Reviewed-on: https://chromium-review.googlesource.com/c/v8/v8/+/4674825
Commit-Queue: Ji Qiu <qiuji@iscas.ac.cn>
Reviewed-by: Ji Qiu <qiuji@iscas.ac.cn>
Cr-Commit-Position: refs/heads/main@{#88776}
---
 .../backend/riscv/instruction-selector-riscv64.cc         | 8 ++------
 1 file changed, 2 insertions(+), 6 deletions(-)

diff --git a/src/compiler/backend/riscv/instruction-selector-riscv64.cc b/src/compiler/backend/riscv/instruction-selector-riscv64.cc
index 2f437174f51..cb03eb2ddd1 100644
--- a/src/compiler/backend/riscv/instruction-selector-riscv64.cc
+++ b/src/compiler/backend/riscv/instruction-selector-riscv64.cc
@@ -2271,10 +2271,7 @@ void InstructionSelectorT<Adapter>::VisitInt64MulWithOverflow(Node* node) {
 }
 
 template <typename Adapter>
-void InstructionSelectorT<Adapter>::VisitWord64Equal(Node* const node) {
-  if constexpr (Adapter::IsTurboshaft) {
-  UNIMPLEMENTED();
-  } else {
+void InstructionSelectorT<Adapter>::VisitWord64Equal(node_t node) {
   if constexpr (Adapter::IsTurboshaft) {
     UNIMPLEMENTED();
   } else {
@@ -2286,11 +2283,10 @@ void InstructionSelectorT<Adapter>::VisitWord64Equal(Node* const node) {
 
     VisitWord64Compare(this, node, &cont);
   }
-  }
 }
 
 template <typename Adapter>
-void InstructionSelectorT<Adapter>::VisitInt64LessThan(Node* node) {
+void InstructionSelectorT<Adapter>::VisitInt64LessThan(node_t node) {
   if constexpr (Adapter::IsTurboshaft) {
   UNIMPLEMENTED();
   } else {
-- 
2.35.1

